src/index.tsx

```diff
@@ -159,16 +161,8 @@ class PyflyByWidget extends Widget {
       p = Promise.resolve(imports);
     }
 
-    const { model } = this._context;
-    // Find the right cell and position to insert cell
-    // //here is where we trigger stuff.
-    const { position, cellIndex } = this._findImportCoordinates();
-    const cell = model.cells.get(cellIndex);
-    const insertIndex = position === -1 ? cell.value.text.length : position;
-    let toInsert = cell.value.text.length === 0 ? imports : `${imports}\n`;
-    if (insertIndex !== 0 && cell.value.text[insertIndex - 1] !== '\n') {
-      toInsert = `\n${toInsert}`;
-    }
+    // creates the cell for imports
+    this._findImportCoordinates();
```

Comment:
While not you, this function name doesn't make clear it modifies state. Maybe `this._findAndSetImportCoordinates();`